01/components/Table/Table.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import TableHeader from './TableHeader';
+import TableBody from './TableBody';
+import TableFooter from './TableFooter';
+
+class Table extends React.Component {
+
+    render() {
+        const {data} = this.props;
+        
+        const onlyItemSummary = data.map(item => {
+            const {price,quantity} =  item;
+            return (
+                {itemSummary: price*quantity}
+            )
+        })
+
+        const dataWithItemSummary = data.map((element, index) => {
+            return { ...element, ...onlyItemSummary[index] };
+          });    
```

Comment:
Osobiście zrobiłbym inaczej:
```
const getSummary = (item) => {....}
const dataWithItemSummary = data.map((element, index) => {
     return { ...element, summary: getSummary(element) };
 });
 ```
 O tyle wydaje mi się czytelniejsze i bezpieczniejsze, że nie korzystamy z indeksu tj. `...onlyItemSummary[index]`. 
 W między czasie może pojawić się kod, który coś może namieszać z indeksami.
 W mojej opcji to nie może nastąpić ;)